mock-api/mock-api-server.ts

```diff
@@ -23,7 +23,23 @@ server.get('/vehicles/*', (req, res) => {
       res.jsonp('Error service unavailable');
       break;
     default:
-      res.jsonp(mockVehicleTecnicalRecordList());
+      // Switch on VIN
+      switch (req.params.id) {
+        case 'delayfound':
+          console.log('Delaying request');
+          setTimeout(() => {res.jsonp(mockVehicleTecnicalRecordList())}, 2500);
+          break;
+        case 'delaynotfound':
+          console.log('Delaying not found request');
+          setTimeout(() => {
+            res.status(500);
+            res.statusMessage = 'Unavailable';
+            res.jsonp('Error service unavailable');
+          }, 2500);
+          break;
+        default:
+          res.jsonp(mockVehicleTecnicalRecordList());
+      }
```

Comment:
I feel like we'd started down a path of command-line switches to affect the mock-api responses, whereas this approach introduces magic search criteria. Could we stick to one approach for simplicity's sake?

Personally I like the command-line approach, but I don't mind the change if there's a compelling reason.

Staying command-line, we could wrap the entire switch statement in a `setTimeout(() => {...}, getDelay())` where `getDelay()` returns the value from the command line switch or return zero if not defined so the response is not delayed.

We could use this same approach for all of the mocked api calls (i.e. test-records), and it keeps the NotFound, and ServiceError logic to a single implementation (the code as-is has 2 different HTTP responses for NotFound - a 404 (line 16) and 500 (line 35)).

@danielchildsbjss - what do you think?

Reply:
The reason I didn't do that is it's a pain to keep changing what you want from the api. It's much easier to leave the service running and just choose what you want. Eg. with the error message stuff with @gjulien-bjss it was much easier to just do `return404` and `return202` commands to check that the error messages were working as expected.

Reply:
Converted the entire PR to this method, awaiting discussion.

Reply:
I'm happy with this approach, but we might start hitting problems if/when we add validation to the VIN search box and these secret commands can't get passed through.
Some of these things I'm not sure we need - if we want to simulate 500 not available why not just close the mock server and make it really unavailable?